### ISSUE TITLE
Add a changelog

### ISSUE THREAD

**Issue by yvele** [dmnd/dedent#48]
Hello, you've recently updated dedent from 0.7.0 to 1.0.2 which means in semver that this new version is major and potentially breaking.

Where can I found your changelog? 

Dedent is massively used https://www.npmjs.com/package/dedent

![image](https://github.com/dmnd/dedent/assets/4225430/b413773d-4f47-49ec-a66e-427c4a93c5d6)

The community will highly benefits from having an easy to find changelog (with a copy within https://github.com/dmnd/dedent/releases).

And also please could you also use GitHub relase -> https://github.com/dmnd/dedent/releases that makes it really easy to find. Specially for developers and maintainers like me that deals with hundreds of packages to updates 🙏 

Thanks for your awesome work

**Comment by yvele** [comment]
At least a manually written one for the v0 -> v1 
what is breaking? what changes?

**Comment by Haroenv** [comment]
I believe the changes are related to the files which are published, but nothing to the functionality

**Comment by JoshuaKGoldberg** [comment]
Hiyall - thanks for raising this! Clearly I underestimated how much folks want a real CHANGELOG. Sorry for the lack of clarity!

As @Haroenv said, there are no breaking functionality changes in 1.0. The full list of changes from 0.x to 1.0 are:

1. Bugfix: #28
2. Mostly build changes, should not change runtime: #38
2. Mostly build changes, should not change runtime: #32
3. Fixed runtime module exports to not have a breaking change: #40

Will add one soon - #36 has a list of steps I'm taking, which include adding automatic generation that cross-posts to GitHub releases. 👍
